k8sdeps/transformer/prune/prune.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prune
+
+import (
+	"sigs.k8s.io/kustomize/k8sdeps/kunstruct"
+	"sigs.k8s.io/kustomize/k8sdeps/transformer/hash"
+	"sigs.k8s.io/kustomize/pkg/gvk"
+	"sigs.k8s.io/kustomize/pkg/resid"
+	"sigs.k8s.io/kustomize/pkg/resmap"
+	"sigs.k8s.io/kustomize/pkg/resource"
+	"sigs.k8s.io/kustomize/pkg/transformers"
+	"sigs.k8s.io/kustomize/pkg/types"
+)
+
+// pruneTransformer compute the ConfigMap used in prune
+type pruneTransformer struct {
+	append      bool
+	cmName      string
+	cmNamespace string
+}
+
+var _ transformers.Transformer = &pruneTransformer{}
+
+// NewPruneTransformer makes a pruneTransformer.
+func NewPruneTransformer(p *types.Prune, append bool) transformers.Transformer {
+	if p == nil || p.Type != "alphaConfigMap" {
```

Comment:
Using `alpha` here as a signal that the prune feature is alpha?

It seems excessive to use that word in the details of the implementation - and it implies we have to change it to beta, GA (or remove it), which means we change the API...

isn't it  sufficient to put `alpha` in the command name, e.g `alpha_prune` instead of `prune`?
Or maybe add an annotation in the configmap that says "aplha:  This is part of an alpha feature!"

or something

Reply:
I see what you mean. Adding `alpha` in the API might not be the best way to notify the feature state. How about removing `alpha` from the prune type
```
type Prune struct {
	Type           string   `json:"type,omitempty" yaml:"type,omitempty"`
	ConfigMap NameArgs `json:"configMap,omitempty" yaml:"configMap,omitempty"`
}
```
and adding "alpha: This is part of an alpha feature!" in the annotation.

Reply:
yes, please do just that